### PR TITLE
New result analysis Script added

### PR DESCRIPTION
**Pull request by ibnyusuf**

### CODE REVIEW

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@ibnyusuf: Could you add some documentation? Especially how to run your script.

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+import sys, copy, random
+
+import cPickle as pickle
+import gzip
+
+def read_strats(data_file_names):
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Add some doc string for this function

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  solvers = {}   # maps solvers to expers where each exper is a {solved_problems:status} dict
+  
+  # loading
+  for name in data_file_names:
+    with open(name,"r") as f:
+      for line in f:
+        _,prob,_,solver,_,config,_,_,time,_,memory,stat = line.split(",")[:12]
+
+        if ".rm" in prob:
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Why?

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          pass
+
+        solver_name = solver+"_"+config
+
+	if "SMT" in prob:
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Why?

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        if prob in exper: # if there were multiple runs of the same strategy, conut
+          pass
+        else:
+          exper[prob] = stat.strip() 
+
+  return solvers
+
+if __name__ == "__main__":
+  
+  solvers = read_strats(["Job42771_info.csv"])
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Could the list of files be passed as an argument?

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  #del solvers["solver_configuration"]
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Remove commented instructions

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    count = 0
+    for (prob,stat) in exper.items(): 
+      if stat=="Refutation" or stat=="Theorem" or stat=="Unsatisfiable" or stat=="unsat":
+        count +=1
+        if prob not in solved:
+          solved[prob] = set()
+        solved[prob].add(solver)
+    print solver,"\t:",count
+             
+  #sys.exit(0)
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Remove commented instructions

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py:1]: Could you force the usage of python3?

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+  for (solver,exper) in sorted(solvers.items()):
+    count = 0
+    for (prob,stat) in exper.items(): 
+      if stat=="Refutation" or stat=="Theorem" or stat=="Unsatisfiable" or stat=="unsat":
+        count +=1
+        if prob not in solved:
+          solved[prob] = set()
+        solved[prob].add(solver)
+    print solver,"\t:",count
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: Convert to python 3 print.

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]
Some comments. Also, it would be great if you could add some tests.

File: scripts/competitions/results_analysis/analyze_esbmc_results.py (hunk 8 above)
Comment (ibnyusuf) [reply scripts/competitions/results_analysis/analyze_esbmc_results.py]: Sorry, I uploaded the wrong script. This is a Vampire script!

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

@ibnyusuf: Could you also create a wiki page in https://github.com/esbmc/esbmc/wiki to describe how to use this script?

File: scripts/competitions/results_analysis/analyze_esbmc_results.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  for b in uniques_1:
+    print("  " + b)
+
+  print("\n\nUniques 2:\n")
+  for b in uniques_2:
+    print("  " + b)
+
+#  print uniques_1
+#  print "\n\n\n"
+#  print uniques_2
```
Comment (rafaelsamenezes) [inline scripts/competitions/results_analysis/analyze_esbmc_results.py]: remove commented instructions

**Review by rafaelsamenezes — APPROVED** [review]
Minor comments. LGTM

File: scripts/competitions/results_analysis/analyze_esbmc_results.py (hunk 9 above)
Comment (ibnyusuf) [reply scripts/competitions/results_analysis/analyze_esbmc_results.py]: Done

**Pull request merged**